lib/vmdb/util.rb

```diff
@@ -1,19 +1,26 @@
 module VMDB
   module Util
-    def self.http_proxy_uri(proxy_config = :default)
+    def self.http_proxy(proxy_config = :default)
       proxy = ::Settings.http_proxy[proxy_config].to_hash
       proxy = ::Settings.http_proxy.to_hash unless proxy[:host]
       return nil if proxy[:host].blank?
 
-      user     = proxy.delete(:user)
-      user &&= CGI.escape(user)
-      password = proxy.delete(:password)
+      proxy
+    end
+
+    def self.http_proxy_uri(proxy_config = :default)
+      proxy = http_proxy(proxy_config)
+      return if proxy.nil?
+
+      user       = proxy.delete(:user)
+      user     &&= CGI.escape(user)
+      password   = proxy.delete(:password)
       password &&= CGI.escape(password)
-      userinfo = "#{user}:#{password}".chomp(":") unless user.blank?
+      userinfo   = "#{user}:#{password}".chomp(":") if user.present?
 
-      proxy[:userinfo]   = userinfo
+      proxy[:userinfo] = userinfo
       proxy[:scheme] ||= "http"
-      proxy[:port] &&= proxy[:port].to_i
+      proxy[:port]   &&= proxy[:port].to_i
```

Comment:
The diff looks complicated but I basically just split up the "return the hash" from the "build a URI"